src/neo-vm/ExecutionEngine.cs

```diff
@@ -324,11 +324,9 @@ private bool ExecuteInstruction()
                         if (!CurrentContext.TryStack.TryPop(out ExceptionHandingContext currentTry))
                             return false;
 
-                        if (currentTry.Rethrow)
-                        {
-                            currentTry.Rethrow = false;
+                        if (!currentTry.HasCatch && currentTry.ExceptionItem != null)
                             return HandleException(currentTry.ExceptionItem);
```

Comment:
We also need to rethrow the exception which happened in catch body.

Reply:
If the exception is thrown in the catch segment, it is not a rethrowing. This situation should be handled by the `THROW` instruction.

Reply:
> This situation should be handled by the THROW instruction.

It's right. But it may in the same `ExceptionHandingContext`, like this example:

```cs
try{ 
   ...
   throw Error1
}catch{
   ....
   throw Error2 // This error need to rethrow in finally segment.
}finally{
   ... // When error2 happened, we need to execute here.
}
```

Reply:
If an exception is thrown in the `catch` segment, and there is no `finally` segment, then the exception should be cached. It shouldn't replace the `ExceptionItem` of the current `ExceptionHandingContext`.

There is also a special case:

```c#
static void Main()
{
    try
    {
        Test();
    }
    catch (Exception ex)
    {
        //The ex should be #3
    }
}

private static void Test()
{
    try
    {
        throw new Exception("#1");
    }
    catch
    {
        throw new Exception("#2");
    }
    finally
    {
        throw new Exception("#3");
    }
}
```

The `#2` exception should be replaced by `#3`.

Reply:
So we don't throw `#2` exception, if there is no `#3` exception in finally segment? 

Or we can rename `ExceptionItem` to `RethrowExceptionItem`, and delete `tryContext.ExceptionItem = exceptionItem` at the beginning of `catch`, since we have already pop the error onto stack.

Reply:
> So we don't throw #2 exception, if there is no #3 exception in finally segment?

No. If there is no #3, #2 is caught by `Main()`. Otherwise, #3 is caught by `Main()`.

Reply:
>If there is no `#3`, `#2` is caught by Main()

If so, we need to throw `#2` at the end of `ENDF`?
```cs
if (!currentTry.HasCatch && currentTry.ExceptionItem != null)
need to change  -> 
if (currentTry.ExceptionItem != null)
```

Reply:
If you change to this, `#1` will be re-thrown. We need to throw `#2`. `#1` has already been handled.

I think we need to add a new field to `ExecutionEngine`. Maybe `ExecutionEngine.UncaughtException`.

Reply:
I prefer to add a `UncaughtException/RethrowException/HangException` field in `ExceptionHandingContext`. As we may need to clear the `UncaughtException` in `ExecutionEngine`, when `#3` exception happened.  Which do you prefer?

Reply:
Add the field `UncaughtException`.

Reply:
Done.